sdk/lib/core/pattern.dart

```diff
@@ -72,6 +72,21 @@ abstract class Match {
   ///
   /// The result may be `null` if the pattern didn't assign a value to it
   /// as part of this match.
+  /// ```dart
+  /// final string = '[00:13.37] This is a chat message.';
+  /// final regExp = RegExp(r'^\[([0-9]+):([0-9]+)\.([0-9]+)\](.*)$');
+  /// final match = regExp.firstMatch(string);
+  /// final message = match?.group(0);
```

Comment:
it would work also without print-lines (shorter example then),  
i would modify it to this style with match group values:
final hours = match?.group(1); // 00
final minutes = match?.group(2);  // 13
...

Reply:
consider also adding: 
final count = match?.groupCount; // 4

after line:
final match = regExp.firstMatch(string);

Reply:
fixed